frontend/components/common/Table.tsx

```diff
@@ -0,0 +1,407 @@
+import { ReactElement, useMemo, useState, useEffect } from 'react'
+import {
+ usePagination,
+ useTable,
+ useFilters,
+ useGlobalFilter,
+ useSortBy,
+} from 'react-table'
+import Select from 'react-select'
+import { Icon } from '.'
+import {
+ BORDER,
+ WHITE,
+ GREEN,
+ HOVER_GRAY,
+ ALLBIRDS_GRAY,
+ CLUBS_GREY,
+ BLUE,
+ FOCUS_GRAY,
+} from '../../constants/colors'
+import { BODY_FONT } from '../../constants/styles'
+import styled from 'styled-components'
+import { MD, mediaMaxWidth, BORDER_RADIUS } from '../../constants/measurements'
+ 
+const styles = {
+ control: ({ background, ...base }) => {
+   return {
+     ...base,
+     border: `1px solid ${BORDER}`,
+     boxShadow: 'none',
+     width: '100%',
+   }
+ },
+ option: ({ background, ...base }, { isFocused, isSelected }) => {
+   const isEmphasized = isFocused || isSelected
+   return {
+     ...base,
+     background: isEmphasized ? FOCUS_GRAY : background,
+     color: CLUBS_GREY,
+   }
+ },
+}
+
+type filterOption = {
+  options : string[],
+  label: string,
+}
+
+type Row= { [key: string]: any }
+
+type Action = {
+  name: string,
+  classes: string,
+  Icon : string,
+  clickFunction : (any) => any
+}
+
+type selectFilter = {
+  value: string,
+  label: string
+}
+ 
+type tableProps = {
+ columns: Row[],
+ data: Row[],
+ searchableColumns: string[]
+ filterOptions: filterOption [],
+ actions?: Action[]
+}
+ 
+const Styles = styled.div`
```

Comment:
I love how you styled this table by hand, but it might be ideal to use the existing [Bulma table styles](https://bulma.io/documentation/elements/table/) to match the look of the rest of the site.

Reusing existing library components instead of making your own can save a lot of time going forward, as you have less code to maintain and easier to modify styling.

Reply:
In this case, you wouldn't want to use CSS rules to mimic the styles of existing buttons, but instead use the appropriate `className` to take advantage of Bulma styles.